### ISSUE TITLE
zpool status hangs after resuming from sleep (suspend-to-ram)

### ISSUE THREAD

**Issue by andreevmipt** [openzfs/zfs#14491]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| Arch
Distribution Version	| -
Kernel Version	| 6.1.9-zen1-2-zen
Architecture	| x86_64
OpenZFS Version	| 2.1.99-1710_g973934b965

### Describe the problem you're observing
zfs hangs after resuming from sleep - effectively reattaching usb hdd enclosure

```
[52717.330983] INFO: task txg_sync:24967 blocked for more than 122 seconds.
[52717.331017]       Tainted: P           OE      6.1.9-zen1-2-zen #1
[52717.331022] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[52717.331025] task:txg_sync        state:D stack:0     pid:24967 ppid:2      flags:0x00004000
[52717.331047] Call Trace:
[52717.331056]  <TASK>
[52717.331074]  __schedule+0xb61/0x1470
[52717.331130]  ? spl_kmem_cache_alloc+0xab/0x920 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.331175]  ? spl_kmem_cache_alloc+0xd2/0x920 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.331190]  schedule+0x5e/0xd0
[52717.331195]  schedule_timeout+0x231/0x380
[52717.331207]  ? __bpf_trace_tick_stop+0x10/0x10
[52717.331224]  io_schedule_timeout+0x50/0x80
[52717.331230]  __cv_timedwait_io+0xc4/0x160 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.331258]  ? swake_up_one+0x70/0x70
[52717.331272]  zio_wait+0x13a/0x370 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.335348]  dmu_buf_hold+0xfb/0x170 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.336313]  zap_contains+0x4e/0x140 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.336788]  spa_sync+0xa35/0xf70 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.337150]  ? spa_txg_history_init_io+0x193/0x1c0 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.337506]  txg_sync_thread+0x1fe/0x390 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.337801]  ? txg_quiesce_thread+0x350/0x350 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.338065]  ? spl_taskq_fini+0x80/0x80 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.338108]  thread_generic_wrapper+0x5b/0x70 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.338129]  kthread+0xdb/0x110
[52717.338156]  ? kthread_complete_and_exit+0x20/0x20
[52717.338163]  ret_from_fork+0x1f/0x30
[52717.338185]  </TASK>
```

```
[52717.338978] INFO: task zpool:1025267 blocked for more than 122 seconds.
[52717.338997]       Tainted: P           OE      6.1.9-zen1-2-zen #1
[52717.339000] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[52717.339005] task:zpool           state:D stack:0     pid:1025267 ppid:2436   flags:0x00000002
[52717.339022] Call Trace:
[52717.339028]  <TASK>
[52717.339034]  __schedule+0xb61/0x1470
[52717.339070]  ? arc_read+0xc1f/0x1620 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.339474]  schedule+0x5e/0xd0
[52717.339498]  schedule_timeout+0x231/0x380
[52717.339507]  ? __bpf_trace_tick_stop+0x10/0x10
[52717.339525]  io_schedule_timeout+0x50/0x80
[52717.339536]  __cv_timedwait_io+0xc4/0x160 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.339584]  ? swake_up_one+0x70/0x70
[52717.339594]  zio_wait+0x13a/0x370 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.339865]  dnode_hold_impl+0x195/0x1200 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.340149]  dmu_buf_hold+0x69/0x170 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.340483]  zap_contains+0x4e/0x140 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.340772]  vdev_rebuild_get_stats+0x76/0x230 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.341084]  vdev_config_generate+0x9d2/0xa60 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.341344]  ? spl_kmem_alloc_impl+0x111/0x1a0 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.341394]  ? slab_post_alloc_hook.constprop.0+0x32/0x240
[52717.341418]  ? __kmem_cache_alloc_node+0x194/0x320
[52717.341426]  vdev_config_generate+0x8a3/0xa60 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.341675]  spa_config_generate+0x239/0x4c0 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.341947]  spa_open_common+0x154/0x430 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.342240]  ? __alloc_pages+0x1db/0x11b0
[52717.342258]  spa_get_stats+0x51/0x3f0 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.342566]  ? __kmem_cache_alloc_node+0x194/0x320
[52717.342581]  ? spl_kmem_alloc+0x111/0x1a0 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.342625]  ? spl_kmem_alloc+0x111/0x1a0 [spl 5d87d68e6e8993ef192370cbe3be1a9ba21b5585]
[52717.342641]  zfs_ioc_pool_stats+0x40/0x90 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.342905]  zfsdev_ioctl_common+0x767/0x9f0 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.343835]  ? __kmalloc_node+0x110/0x1a0
[52717.343857]  zfsdev_ioctl+0x53/0xe0 [zfs be95b2199d25f8118d3bdb1dd34ae5c0038c08bc]
[52717.344153]  __x64_sys_ioctl+0x91/0xd0
[52717.344171]  do_syscall_64+0x5c/0x90
[52717.344197]  ? do_user_addr_fault+0x1e9/0x6c0
[52717.344219]  ? sched_clock_cpu+0xd/0xb0
[52717.344229]  ? exc_page_fault+0x74/0x170
[52717.344241]  entry_SYSCALL_64_after_hwframe+0x63/0xcd
[52717.344257] RIP: 0033:0x7fcf2b22c53f
[52717.344529] RSP: 002b:00007fff49d6e230 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
[52717.344538] RAX: ffffffffffffffda RBX: 000055d9ac714270 RCX: 00007fcf2b22c53f
[52717.344543] RDX: 00007fff49d6e2b0 RSI: 0000000000005a05 RDI: 0000000000000003
[52717.344545] RBP: 00007fff49d718a0 R08: 0000000000000000 R09: 0000000000000000
[52717.344548] R10: 000055d9ac724270 R11: 0000000000000246 R12: 00007fff49d6e2b0
[52717.344550] R13: 000055d9ac714270 R14: 000055d9ac6e74e0 R15: 00007fff49d718c4
[52717.344557]  </TASK>
```

### Describe how to reproduce the problem
I have external usb hdd enclosure with 4 disks attached to notebook
pool imported with -d /dev/disk/by-path
On resuming from sleep disks become available only after some time so I got - that's ok
[52507.922643] zio pool=ztnk vdev=/dev/disk/by-path/pci-0000:00:14.0-usb-0:1.3:1.0-scsi-3:0:0:3-part3 error=5 type=1 offset=1719239630848 size=4096 flags=1575040
[52507.922790] zio pool=ztnk vdev=/dev/disk/by-path/pci-0000:00:14.0-usb-0:1.3:1.0-scsi-3:0:0:0-part3 error=5 type=1 offset=1719239630848 size=4096 flags=1575040
[52507.923070] zio pool=ztnk vdev=/dev/disk/by-path/pci-0000:00:14.0-usb-0:1.3:1.0-scsi-3:0:0:4-part3 error=5 type=1 offset=842125266944 size=4096 flags=1575040
[52507.923169] zio pool=ztnk vdev=/dev/disk/by-path/pci-0000:00:14.0-usb-0:1.3:1.0-scsi-3:0:0:2-part3 error=5 type=1 offset=842125266944 size=4096 flags=1575040
[52507.924877] WARNING: Pool 'ztnk' has encountered an uncorrectable I/O failure and has been suspended.

Then disks become available, but pool does not resume, all commands simply hangs (zpool status ...)

**Comment by Mitmischer** [comment]
Maybe a duplicate of #14426. What do `strace` and `/proc/spl/kstat/zfs/dbgmsg` look like for you?
